lib/exceptions.rb

```diff
@@ -170,6 +170,7 @@ class UpgradeFailed < MachineryError; end
     class BinaryDiffError < MachineryError; end
 
     class SshConnectionFailed < MachineryError; end
+    class SshConnectionDisrupted < SystemExit; end
```

Comment:
No this can cause other functions to fail because they depend on data that should have been gathered.
* I meant i already tried it with RuntimeException but i get a backtrace when i use that.
I also thought about taking something different than SystemExit but i would probably need to look at the code that suppresses exceptions while inspecting and that would probably be too much for this card.